analyses/cell-type-wilms-tumor-06/notebook_template/03_clustering_exploration.Rmd

```diff
@@ -0,0 +1,618 @@
+    print(tmp)
+  },
+  error = {
+    expr= print("feature not found")
+    }
+ )
+}
+```
+
+
+
+### Look at specific pathways
+
+#### TP53 pathway
+
+here we will calculate a TP53 score using AddMduleScore and the genes of the HALLMARK_P53_PATHWAY gene set.
+
+```{r fig.height=2.5, fig.width=8, warning=FALSE, out.width='100%'}
+srat <- MSigDB_score(object = srat , category = "H", gs_name = "HALLMARK_P53_PATHWAY", name = "TP53_score")
+visualize_metadata(srat, meta = "TP53_score1", group.by = "seurat_clusters" )
+```
+
+
+#### DNA repair pathway
+
+here we will calculate a DNA_repair score using AddMduleScore and the genes of the HALLMARK_DNA_REPAIR gene set.
+
+```{r fig.height=2.5, fig.width=8, warning=FALSE, out.width='100%'}
+srat <- MSigDB_score(object = srat , category = "H", gs_name = "HALLMARK_DNA_REPAIR", name = "DNA_repair_score")
+visualize_metadata(srat, meta = "DNA_repair_score1", group.by = "seurat_clusters" )
+```
+
+Note: Chemo-treated samples should have higher DNA-damage scores.
+
+
+#### DROSHA target genes
+
+```{r fig.height=2.5, fig.width=8, warning=FALSE, out.width='100%'}
+srat <- MSigDB_score(object = srat , category = "C3", gs_name = "DROSHA_TARGET_GENES", name = "DROSHA_score")
+visualize_metadata(srat, meta = "DROSHA_score1", group.by = "seurat_clusters" )
+```
+
+
+#### DICER1 target genes
+
+
+```{r fig.height=2.5, fig.width=8, warning=FALSE, out.width='100%'}
+srat <- MSigDB_score(object = srat , category = "C3", gs_name = "DICER1_TARGET_GENES", name = "DICER1_score")
+visualize_metadata(srat, meta = "DICER1_score1", group.by = "seurat_clusters" )
+```
+
+
+### Find marker genes for each of the seurat clusters
+
+In addition to the list of known marker genes, we used an unbiased approach to find transcripts that characterized the different clusters.
+We run DElegate::FindAllMarkers2 to find markers of the different clusters and manually check if they do make sense. 
+DElegate::FindAllMarkers2 is an improved version of Seurat::FindAllMarkers based on pseudobulk differential expression method. 
+Please check the preprint from Chistoph Hafemeister: https://www.biorxiv.org/content/10.1101/2023.03.28.534443v1
+and tool described here: https://github.com/cancerbits/DElegate 
+
+Of note, we won't use it for annotation, this is just here to get an idea!
+
+```{r fig.width=10, fig.height=10, out.width='100%'}
+feature_conversion <- srat@assays$RNA@meta.data
+de_results   <- DElegate::FindAllMarkers2(srat, group_column = "seurat_clusters")
+#filter the most relevant markers
+s.markers <- de_results[de_results$padj < params$padj_thershold & de_results$log_fc > params$lfc_threshold & de_results$rate1 > params$rate1_threshold,]
+# add gene symbol for easiest interpretation of the result
+s.markers$gene_ids <- s.markers$feature
+s.markers <- left_join(s.markers,feature_conversion, by = c( "gene_ids") )
+identical(s.markers$feature, s.markers$gene_ids)     # check the quality of the merge, must be true
```

Comment:
Let's turn this into a _check_ instead of just printing. If this is FALSE, we'd like to know about it!

This will cause an error ("stop if not true") if the statement is not true, and print that error message.

```suggestion
stopifnot(
  "Error joining gene ids and feature names" = identical(s.markers$feature, s.markers$gene_ids)   
)
```